### DISCUSSION TITLE
incomplete database install: ER_BAD_FIELD_ERROR: Unknown column 'accountability' in 'field list' after  database install

### DISCUSSION THREAD

**Discussion by terion-name** [directus/directus#13917] [Archive]
### Describe the Bug

Running directus in docker with migrations to setup some basic data. As by migrations docs set the cmd like this:
```
CMD npx directus database install ; npx directus schema apply ./schema.yml -y && npx directus database migrate:latest && npx directus start
```

Getting error at schema apply stage:
```
2022-06-15T20:26:48.140612681Z 20:26:48 🚨 select `collection`, `singleton`, `note`, `sort_field`, `accountability` from `directus_collections` - ER_BAD_FIELD_ERROR: Unknown column 'accountability' in 'field list'
2022-06-15T20:26:48.140643384Z Error: ER_BAD_FIELD_ERROR: Unknown column 'accountability' in 'field list'
```

Database after install has 15 directus tables, which is less then should be:
```
directus_activity
directus_collections
directus_fields
directus_files
directus_folders
directus_migrations
directus_permissions
directus_presets
directus_relations
directus_revisions
directus_roles
directus_sessions
directus_settings
directus_users
directus_webhooks
```

### To Reproduce

see description

### What version of Directus are you using?

9.12.2

### What version of Node.js are you using?

16

### What database are you using?

mysql

### What browser are you using?

brave

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
> As by migrations docs set the cmd like this:

What migration docs? If you're referring to https://docs.directus.io/extensions/migrations/#custom-migrations, that's for custom migration files (using knex migrations in JS), not schema migrations using the schema files. 

In this particular case, the error is thrown because the initial install of the database doesn't contain any of the migrations (and therefore reflects the database of v9.0.0 effectively). In this particular case, you're probably looking to set the command to 

`CMD npx directus bootstrap ; npx directus schema apply ./schema.yml -y && npx directus start`

**Reply by terion-name** [reply]
@rijkvanzanten this does not resolves the issue and closing it beforehand is not ok. I'm referring exactly that migrations docs. I have custom migrations in files that should be run after schema apply. `npx directus bootstrap` will run custom migrations but this should not be done before schema applying. However I can't run them after because of  described behaviour.

And instructions from docs simply do not work — because `schema apply` fails after `database install`

It is expected that `npx directus database install` will install all the system tables and fields — it's kinda logical and again docs state this with provided examlpe. But it does only part of the process what is strange.

If some system migration must be run separately, how can I run them after database install and before schema apply?

**Reply by terion-name** [reply]
what I've made to overcome this was a hacky script `setup.sh`:
```
mv extensions/migrations extensions/migrations_
npx directus bootstrap
mv extensions/migrations_ extensions/migrations
npx directus schema apply ./schema.yml -y
npx directus database migrate:latest
```
and `CMD ./setup.sh && npx directus start`

But obviously it is a very bad approach and something needs to be done to process, especially when even example [from docs](https://docs.directus.io/extensions/migrations/#migrations-and-directus-schema) just can't work

**Reply by piotr-cz** [reply]
As @terion-name mentions, app migrations must run *before* snapshot is applied.
This limits option to hydrate/ seed custom tables with migrations.

**Comment by RANSoftSS** [comment]
@terion-name are there any updates?
